pkg/manager/impl/execution_manager.go

```diff
@@ -184,12 +182,49 @@ func (m *ExecutionManager) addPluginOverrides(ctx context.Context, executionID *
 	return nil, nil
 }
 
-// TODO: Delete this code usage after the flyte v0.17.0 release
+// defaults should be a coalesce of task defaults, and platform defaults.
+// task limits should be the limits from the task, coalesced with the defaults from step one
+// then both should be limited by any platform limits.
+// anything 0 or empty is not set.
+// if both requests and limits end up empty, return nil. if one is empty, return nil for it
+func (m *ExecutionManager) getResources(ctx context.Context, taskResources *core.Resources, platformResources workflowengineInterfaces.TaskResources) *core.Resources {
+
+	// requests: coalesce(task request, platform default)
+	// limits: coalesce(task limits, task requests)
+	// check that defaults and limits are both below platform limit
+	var requestSet runtimeInterfaces.TaskResourceSet
+	var limitSet runtimeInterfaces.TaskResourceSet
+	if taskResources != nil && taskResources.GetRequests() != nil {
+		requestSet = util.GetTaskResourcesAndCoalesce(ctx, taskResources.GetRequests(), platformResources.Defaults)
+	} else {
+		requestSet = platformResources.Defaults
+	}
+	if taskResources != nil && taskResources.GetLimits() != nil {
+		limitSet = util.GetTaskResourcesAndCoalesce(ctx, taskResources.GetLimits(), requestSet)
+	} else {
+		limitSet = requestSet
+	}
+	adjustedRequestSet := util.ConstrainTaskResourceSet(ctx, requestSet, platformResources.Limits)
+	adjustedLimitSet := util.ConstrainTaskResourceSet(ctx, limitSet, platformResources.Limits)
+
+	// convert the sets back to core.Resources
+	requestEntries := util.ConvertTaskResourceSetToCoreResources(adjustedRequestSet)
+	limitEntries := util.ConvertTaskResourceSetToCoreResources(adjustedLimitSet)
+	if len(requestEntries) == 0 && len(limitEntries) == 0 {
+		return nil
+	}
+	res := core.Resources{}
```

Comment:
i think we solve this with more testing.  looked through the code and i don't think it's used.  Also ran locally with no admin config, no matchable resources, and no task resources, and things seem to be running almost correctly.  The pod is still being created with
```
    resources:
      limits:
        cpu: "0"
        memory: "0"
      requests:
        cpu: "0"
        memory: "0"
```
but the fly crd has node resources
```
resources: {}
```

and has execution config task resources:
```
executionConfig:
  Interruptible: null
  MaxParallelism: 25
  OverwriteCache: false
  RecoveryExecution: {}
  TaskPluginImpls: {}
  TaskResources:
    Limits:
      CPU: "0"
      EphemeralStorage: "0"
      GPU: "0"
      Memory: "0"
      Storage: "0"
    Requests:
      CPU: "0"
      EphemeralStorage: "0"
      GPU: "0"
      Memory: "0"
      Storage: "0"
```
which is correct because these are non-nullable.  and resources doesn't show up at all in the task template part of the task definition.

if downstream code can't handle a nil for some reason it should handle it there i feel.